docs/en/upload.md

```diff
@@ -62,11 +62,12 @@ print("Connecting to WiFi access point...")
 wifi.setmode(wifi.STATION)
 wifi.sta.config(SSID, PASSWORD)
 -- wifi.sta.connect() not necessary because config() uses auto-connect=true by default
-tmr.alarm(1, 1000, 1, function()
+wifitimer = tmr.create()
+wifitimer:alarm(1000, tmr.ALARM_AUTO, function()
     if wifi.sta.getip() == nil then
         print("Waiting for IP address...")
     else
-        tmr.stop(1)
+        wifitimer:stop()
```

Comment:
This is an interesting example. Does the wifitimer object get gc'ed? I suspect that the callback holds a reference to the timer and hence it will never be GC'ed. It is that same (anti-)pattern that causes leakages in sockets in the net module.

I wondered whether the callback was invoked with the timer object as an argument -- but the documentation for `register` doesn't really describe the `func` argument.... I suspect that (like with the network callbacks), the func argument should be called with the timer object as the first parameter.... This makes it easy to write code that doesn't retain the extra reference.